flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/FileStoreWriteImpl.java

```diff
@@ -149,7 +149,7 @@ public RecordWriter createCompactWriter(
             ExecutorService compactExecutor,
             List<DataFileMeta> restoredFiles) {
         return new CompactWriter(
-                CompactUnit.fromFiles(options.numLevels - 1, restoredFiles),
+                CompactUnit.fromFiles(getRestoreMaxLevel(restoredFiles) - 1, restoredFiles),
```

Comment:
We can use `Levels` to produce `CompactUnit` by `CompactUnit.fromLevelRuns`.
In this way, we don't need to change `Levels`.
And we should add a test case here.